prow/plugins/lgtm/lgtm.go

```diff
@@ -202,3 +223,34 @@ func handlePullRequest(gc ghLabelClient, pe github.PullRequestEvent, log *logrus
 	}
 	return nil
 }
+
+func loadRepoOwners(gc githubClient, ownersClient *repoowners.Client, org, repo string, number int) (*repoowners.RepoOwners, error) {
+	pr, err := gc.GetPullRequest(org, repo, number)
+	if err != nil {
+		return nil, err
+	}
+	return ownersClient.LoadRepoOwners(org, repo, pr.Base.Ref)
+}
+
+// getChangedFiles returns all the changed files for the provided pull request.
+func getChangedFiles(gc githubClient, org, repo string, number int) ([]string, error) {
+	changes, err := gc.GetPullRequestChanges(org, repo, number)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get PR changes for %s/%s#%d", org, repo, number)
+	}
+	var filenames []string
+	for _, change := range changes {
+		filenames = append(filenames, change.Filename)
+	}
+	return filenames, nil
+}
+
+// loadReviewers returns all reviewers and approvers from all OWNERS files that
+// cover the provided filenames.
+func loadReviewers(ro *repoowners.RepoOwners, filenames []string) sets.String {
+	reviewers := sets.String{}
+	for _, filename := range filenames {
+		reviewers = reviewers.Union(ro.Approvers(filename)).Union(ro.Reviewers(filename))
```

Comment:
nit: `reviewers.Insert` is more efficient.

Reply:
Not sure how can I use Insert w/o recreating Union since both Approvers and Reviewers return sets instead of strings.